nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/dto/EntityFactory.java

```diff
@@ -77,6 +77,8 @@ public ProcessorEntity createProcessorEntity(final ProcessorDTO dto, final Revis
         if (dto != null) {
             entity.setPermissions(permissions);
             entity.setStatus(status);
+            status.setCanRead(permissions.getCanRead());
```

Comment:
Yup, hit it when I was trying to delete a connection:

2016-07-28 21:41:44,415 INFO [NiFi Web Server-231] org.apache.nifi.web.filter.RequestLogger Attempting request for (anonymous) DELETE http://localhost:8080/nifi-api/connections/344c598c-0156-1000-0000-000053115355 (source ip: 127.0.0.1)
2016-07-28 21:41:44,429 ERROR [NiFi Web Server-231] o.a.nifi.web.api.config.ThrowableMapper An unexpected error has occurred: java.lang.NullPointerException. Returning Internal Server Error response.
java.lang.NullPointerException: null
        at org.apache.nifi.web.api.dto.EntityFactory.createConnectionEntity(EntityFactory.java:251) ~[classes/:na]
        at org.apache.nifi.web.StandardNiFiServiceFacade.deleteConnection(StandardNiFiServiceFacade.java:858) ~[classes/:1.0.0-SNAPSHOT]
        at org.apache.nifi.web.StandardNiFiServiceFacade$$FastClassBySpringCGLIB$$358780e0.invoke(<generated>) ~[classes/:1.0.0-SNAPSHOT]